runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java

```diff
@@ -5,4 +5,17 @@
 public interface Machine {
 
     long[] call(int funcId, long[] args) throws ChicoryException;
+
+    @FunctionalInterface
+    interface Operands {
+        long get(int index);
+    }
+
+    static int readMemPtr(MStack stack, Operands operands) {
```

Comment:
Make this a static protected method on `InterpreterMachine`. It doesn't belong on `Machine` as it's only useful for the interpreter (since it uses `MStack`).

Reply:
good catch, I'll move it back